### PR TITLE
Updated the terms of service.

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/328, https://github.com/mitodl/micromasters/issues/280
#### What's this PR do?
- Update the existing terms of service.

#### Where should the reviewer start?
- See page TermsOfServicePage.js

#### How should this be manually tested?
- Checkout page ``/terms_of_service/``

@pdpinch @aliceriot @nicolaeherrera 

#### Screenshots (if appropriate)
screen shot 2016-05-24 at 3 09 11 am

### CODE REVIEW

**Comment by giocalitri** [comment]
@amir-qayyum-khan There is a small linting error.
Please fix

**Comment by amir-qayyum-khan** [comment]
Done @giocalitri

File: static/js/containers/TermsOfServicePage.js [hunk 1]
```diff
@@ -31,19 +31,106 @@ class TermsOfServicePage extends React.Component {
...
+    );
     return <div>
       <h3>
         MIT Micromasters Terms of Service
       </h3>
 
       <p>
-        Please read through the terms of service. To enroll in MIT Micromasters you
-        must agree to the terms of service.
+        Welcome to the MITx MicroMaster's website (the “Site”).  By accessing this Site, users agree to be
```
Comment (pdpinch) [inline static/js/containers/TermsOfServicePage.js]: A question and a comment:

1. Do we need to escape the single quote? GitHub's syntax checker seems upset. 
2. Let's use straight quotes, i.e. 

```
(the "Site").
```

**Comment by amir-qayyum-khan** [comment]
@pdpinch @noisecapella can you review this.

File: static/js/containers/TermsOfServicePage.js [hunk 2]
```diff
@@ -31,25 +36,147 @@ class TermsOfServicePage extends React.Component {
...
+      </Grid>
+      <div className="card-copy terms-of-service-main">
+        <div className="program">
+          <div className="terms-card-header">
+            Terms of service
+          </div>
+          <div className="course-list terms-of-service-container">
+            <div className="course-list-header course-list-center course-list-row"
+              style={{height: DASHBOARD_COURSE_HEIGHT}}>
+              <h4 className="course-title">Please read and agree to our Terms of Service</h4>
```
Comment (pdpinch) [inline static/js/containers/TermsOfServicePage.js]: This isn't going to be the right className. We seem to be all over the place with these right now. 

The styling should be the same as https://github.com/mitodl/micromasters/blob/master/static/scss/dashboard.scss#L212-L214

Let me suggest terms-of-service-title for consistency, although I think these might all be the same style.

File: static/scss/dashboard.scss [hunk 3]
```diff
@@ -251,3 +251,25 @@ c.validation-wrapper {
...
+    font-weight: bold;
+  }
+  .terms-of-service-container {
+    height: 500px;
+    overflow-y: scroll;
+    li {
+      display: list-item;
+    }
+    .terms-of-service-content {
+      padding: 7px;
```
Comment (pdpinch) [inline static/scss/dashboard.scss]: Padding here should be similar to title, 20px I think? 

here's a screenshot to help explain: 
screenshot-localhost 8079 2016-05-20 08-24-18

File: static/scss/dashboard.scss [hunk 4]
```diff
@@ -251,3 +251,25 @@ c.validation-wrapper {
...
+    }
+  }
+}
```
Comment (pdpinch) [inline static/scss/dashboard.scss]: add newline

File: static/scss/dashboard.scss [hunk 5]
```diff
@@ -251,3 +251,25 @@ c.validation-wrapper {
...
+  margin-left: auto;
+  margin-right: auto;
+  div.terms-card-header {
+    padding: 25px 0 25px 0;
+    font-size: x-large;
+    color: #b34a4c;
+    font-weight: bold;
+  }
+  .terms-of-service-container {
+    height: 500px;
```
Comment (pdpinch) [inline static/scss/dashboard.scss]: This is too tall. How about 250px, which I think is shorter than the mock-ups.

File: static/js/containers/TermsOfServicePage.js [hunk 6]
```diff
@@ -2,8 +2,13 @@
 import React from 'react';
 import Button from 'react-bootstrap/lib/Button';
 import { connect } from 'react-redux';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+import UserImage from '../components/UserImage';
 
 import { saveProfile, FETCH_SUCCESS } from '../actions';
+import {
+  DASHBOARD_COURSE_HEIGHT,
+} from '../constants';
```
Comment (pdpinch) [inline static/js/containers/TermsOfServicePage.js]: this is the right height to use, but it seems weird to me to reuse the style in this context. This isn't the dashboard. This isn't a course.

File: static/js/containers/TermsOfServicePage.js [hunk 7]
```diff
@@ -31,25 +36,147 @@ class TermsOfServicePage extends React.Component {
...
-        Please read through the terms of service. To enroll in MIT Micromasters you
-        must agree to the terms of service.
-      </p>
+    const profile = this.props.userProfile.profile;
+    let linkToEdxTermsOfService = (
+      <a href="https://www.edx.org/edx-terms-service" target='_blank'>https://www.edx.org/edx-terms-service</a>
+    );
+    let imageUrl = `${SETTINGS.edx_base_url}/static/images/profiles/default_120.png`.
+    //replacing multiple "/" with a single forward slash, excluding the ones following the colon
+    replace(/([^:]\/)\/+/g, "$1");
```
Comment (pdpinch) [inline static/js/containers/TermsOfServicePage.js]: What's this for?

**Comment by pdpinch** [comment]
I made some comments about the styling, but I'd like @noisecapella to review for consistently with the rest of the profile.

File: static/js/containers/TermsOfServicePage.js (hunk 7 above)
Comment (noisecapella) [inline static/js/containers/TermsOfServicePage.js]: It removes duplicate slashes from the user image URL. But, this code already exists elsewhere in the project and we should create a component to share it if we are using it twice

File: static/js/containers/TermsOfServicePage.js (hunk 6 above)
Comment (noisecapella) [inline static/js/containers/TermsOfServicePage.js]: I agree, the CSS classes used in the terms of service should be specific to the terms of service, or general to the entire dashboard piece of the site

File: static/scss/dashboard.scss [hunk 8]
```diff
@@ -251,3 +251,25 @@ c.validation-wrapper {
 .education-icons {
   text-align: right;
 }
+
+.terms-of-service-main {
+  width: 75%;
+  margin-left: auto;
+  margin-right: auto;
+  div.terms-card-header {
+    padding: 25px 0 25px 0;
```
Comment (noisecapella) [inline static/scss/dashboard.scss]: If you're going to have the same horizontal and vertical padding you should format it like this: `padding: 25px 0;`

**Comment by amir-qayyum-khan** [comment]
After Update now page look like:

screen shot 2016-05-24 at 3 09 11 am

**Comment by amir-qayyum-khan** [comment]
@pdpinch @noisecapella please review. 
screen shot 2016-05-24 at 3 09 11 am

File: static/js/constants.js [hunk 9]
```diff
@@ -231,5 +231,5 @@ export const DASHBOARD_RESPONSE = [
 
 export const DATE_FORMAT = 'YYYY-MM-DD';
 
-export const DASHBOARD_COURSE_HEIGHT = 70;
```
Comment (noisecapella) [inline static/js/constants.js]: Please make a separate constant. We don't necessarily want the two numbers to be the same.

File: static/js/containers/TermsOfServicePage.js [hunk 10]
```diff
@@ -31,25 +36,149 @@ class TermsOfServicePage extends React.Component {
   }
 
   render() {
-    return <div>
-      <h3>
-        MIT Micromasters Terms of Service
-      </h3>
+    const profile = this.props.userProfile.profile;
+    let linkToEdxTermsOfService = (
+      <a href="https://www.edx.org/edx-terms-service" target='_blank'>https://www.edx.org/edx-terms-service</a>
```
Comment (noisecapella) [inline static/js/containers/TermsOfServicePage.js]: Can you inline this variable?

**Comment by noisecapella** [comment]
Would you mind waiting until #398 gets merged? It will include a header component which we should use here also.

**Comment by pdpinch** [comment]
There's still not enough padding/margin between the text and the box border. See https://github.com/mitodl/micromasters/pull/346/commits/8d20cd8bd0b63b5d67df67ad786d08a284f28d6a#r64032885

**Comment by noisecapella** [comment]
@amir-qayyum-khan #398 has been merged, there is a new `Jumbotron` class which was added which should take care of the header for you

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella @pdpinch , can i merge it?

File: static/js/containers/TermsOfServicePage.js [hunk 11]
```diff
@@ -31,25 +35,130 @@ class TermsOfServicePage extends React.Component {
...
+                  Welcome to the MITx MicroMaster’s website (the “Site”).  By accessing this Site, users agree to be
+                  bound by the following terms and conditions which MIT may revise at any time.  Users are encouraged
+                  to visit this page periodically to review current terms and conditions, as your continued use of this
+                  Site signifies your agreement to these term and conditions.  If you do not understand or do not agree
+                  to be bound by these terms and conditions, please exit this Site immediately.
+                </p>
+                <ol>
+                  <li>
+                    Your use of this Site is entirely voluntary.  This Site is designed to assist and facilitate your
+                    successful attainment of the MITx Micromasters credential, but use of this Site is not required
```
Comment (pdpinch) [inline static/js/containers/TermsOfServicePage.js]: spelling: `MicroMaster’s`

**Comment by pdpinch** [comment]
One minor spelling change (thinking ahead to #236). 

I'm satisfied. @noisecapella should take another quick look.

File: static/js/containers/TermsOfServicePage.js [hunk 12]
```diff
@@ -31,25 +35,130 @@ class TermsOfServicePage extends React.Component {
...
-      </p>
-
-      <Button
-        bsStyle="success"
-        onClick={this.handleAgree.bind(this)}
-      >Agree</Button> <Button bsStyle="danger" href="/logout">Cancel</Button>
-    </div>;
+    const profile = this.props.userProfile.profile;
+    let preferredName = profile.preferredName || SETTINGS.name;
+    let linkToEdxTermsOfService = <a href="https://www.edx.org/edx-terms-service" target='_blank'>
```
Comment (noisecapella) [inline static/js/containers/TermsOfServicePage.js]: Can you inline this variable? There doesn't seem to be a reason to have this outside the rest of the HTML below

File: static/js/containers/TermsOfServicePage.js [hunk 13]
```diff
@@ -31,25 +35,130 @@ class TermsOfServicePage extends React.Component {
...
-      <p>
-        Add terms of service here
-      </p>
-
-      <Button
-        bsStyle="success"
-        onClick={this.handleAgree.bind(this)}
-      >Agree</Button> <Button bsStyle="danger" href="/logout">Cancel</Button>
-    </div>;
+    const profile = this.props.userProfile.profile;
```
Comment (aliceriot) [inline static/js/containers/TermsOfServicePage.js]: you could destructure this:

```js
const { userProfile: { profile } } = this.props;
```

File: static/js/containers/TermsOfServicePage.js [hunk 14]
```diff
@@ -2,8 +2,12 @@
 import React from 'react';
 import Button from 'react-bootstrap/lib/Button';
 import { connect } from 'react-redux';
+import Jumbotron from '../components/Jumbotron';
 
 import { saveProfile, FETCH_SUCCESS } from '../actions';
+import {
+  TERMS_CARD_ROW_HEIGHT,
+} from '../constants';
```
Comment (aliceriot) [inline static/js/containers/TermsOfServicePage.js]: can you make this one line?

**Comment by amir-qayyum-khan** [comment]
Done @pdpinch @noisecapella @aliceriot

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**